x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -85,28 +92,76 @@ protected void masterOperation(
-                LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                enabled,
-                Collections.emptyMap()
+            listener.onResponse(
+                new XPackUsageFeatureResponse(
+                    new EnterpriseSearchFeatureSetUsage(
+                        enabled,
+                        LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
+                        searchApplicationsUsage
+                    )
+                )
             );
-            listener.onResponse(new XPackUsageFeatureResponse(usage));
+        });
+
+        ListSearchApplicationAction.Request searchApplicationsCountRequest = new ListSearchApplicationAction.Request(
+            null,
+            new PageParams(0, 10_000)
+        );
+        clientWithOrigin.execute(ListSearchApplicationAction.INSTANCE, searchApplicationsCountRequest, searchApplicationsCountListener);
+    }
+
+    private void addSearchApplicationsUsage(ListSearchApplicationAction.Response response, Map<String, Object> searchApplicationsUsage) {
+        searchApplicationsUsage.put(EnterpriseSearchFeatureSetUsage.COUNT, response.queryPage().count());
+
+        List<Map<String, Object>> searchApplications = new ArrayList<>();
+
+        for (SearchApplicationListItem searchApplication : response.queryPage().results()) {
+            Map<String, Object> usage = new HashMap<>();
+            usage.put(EnterpriseSearchFeatureSetUsage.NAME, searchApplication.name());
+
+            addSearchApplicationSchemaStats(searchApplication, usage);
+
+            searchApplications.add(usage);
+        }
+
+        searchApplicationsUsage.put(EnterpriseSearchFeatureSetUsage.SEARCH_APPLICATIONS, searchApplications);
+    }
+
+    private void addSearchApplicationSchemaStats(SearchApplicationListItem searchApplication, Map<String, Object> searchApplicationUsage) {
+        try {
+            FieldCapabilitiesResponse fieldCapsResp = clientWithOrigin.prepareFieldCaps(searchApplication.indices())
+                .setFields("*")
+                .execute()
+                .get();
+            Map<String, Map<String, FieldCapabilities>> fieldCaps = fieldCapsResp.get();
+
+            int totalSchemaFields = fieldCaps.size();
+            int totalSchemaFieldConflicts = 0;
+            for (Map<String, FieldCapabilities> field : fieldCaps.values()) {
+                if (field.size() > 1) {
+                    totalSchemaFieldConflicts++;
+                }
+            }
+
+            searchApplicationUsage.put(EnterpriseSearchFeatureSetUsage.SCHEMA_FIELD_COUNT, totalSchemaFields);
+            searchApplicationUsage.put(EnterpriseSearchFeatureSetUsage.SCHEMA_FIELD_CONFLICT_COUNT, totalSchemaFieldConflicts);
+        } catch (ExecutionException | InterruptedException e) {
+            logger.warn("Failed to get search application schema field counts in Enterprise Search usage", e);
```

Comment:
I think we want to handle failure in the listener as well as log

Reply:
i'm sorry i don't understand. could you explain this further, please?